tests/unit/via/services/pdf_url_test.py

```diff
@@ -0,0 +1,137 @@
+        svc,
+        quantized_expiry,
+    ):
+        signed_url = svc.sign_url(
+            "https://example.com/foo/bar.pdf?q=s", "/proxy/static/"
+        )
+
+        quantized_expiry.assert_called_once_with(max_age=timedelta(hours=25))
+        signed_url_parts = signed_url.split("/")
+        signature = signed_url_parts[5]
+        expiry = signed_url_parts[6]
+        assert signature == "qTq65RXvm6P2Y4bfzWdPzg"
+        assert expiry == "1581183021"
+
+    @pytest.fixture
+    def svc(self, pyramid_settings):
+        return _NGINXSigner(
+            pyramid_settings["nginx_server"],
+            pyramid_settings["nginx_secure_link_secret"],
+        )
+
+    @pytest.fixture(autouse=True)
+    def quantized_expiry(self, patch):
+        return patch(
+            "via.services.pdf_url.quantized_expiry",
+            return_value=datetime(
+                year=2020,
+                month=2,
+                day=8,
+                hour=17,
+                minute=30,
+                second=21,
+                tzinfo=timezone.utc,
+            ),
+        )
+
+
+class TestPDFURLBuilder:
+    @pytest.mark.parametrize(
+        "file_details,url",
+        (
+            (
+                {"file_id": "FILE_ID"},
+                "http://example.com/google_drive/FILE_ID/proxied.pdf"
+                "?url=http%3A%2F%2Fgdrive%2Fdocument.pdf",
+            ),
+            (
+                {"file_id": "FILE_ID", "resource_key": None},
+                "http://example.com/google_drive/FILE_ID/proxied.pdf"
+                "?url=http%3A%2F%2Fgdrive%2Fdocument.pdf",
+            ),
+            (
+                {"file_id": "FILE_ID", "resource_key": "RESOURCE_KEY"},
+                "http://example.com/google_drive/FILE_ID/RESOURCE_KEY/proxied.pdf"
+                "?url=http%3A%2F%2Fgdrive%2Fdocument.pdf",
+            ),
+        ),
+    )
+    def test_google_file_url(
+        self, svc, google_drive_api, secure_link_service, file_details, url
+    ):
+        google_drive_api.parse_file_url.return_value = file_details
+
+        pdf_url = svc.get_pdf_url("http://gdrive/document.pdf")
+
+        secure_link_service.sign_url.assert_called_once_with(url)
+        assert pdf_url == secure_link_service.sign_url.return_value
+
+    def test_nginx_file_url(self, google_drive_api, svc):
+        google_drive_api.parse_file_url.return_value = None
+
+        pdf_url = svc.get_pdf_url("http://nginx/document.pdf")
+
+        svc.nginx_signer.sign_url.assert_called_once_with(
+            "http://nginx/document.pdf", nginx_path="/proxy/static/"
+        )
+        assert pdf_url == svc.nginx_signer.sign_url.return_value
+
+    @pytest.fixture
+    def svc(
+        self,
+        google_drive_api,
+        secure_link_service,
+        pyramid_request,
+        NGINXSigner,
+    ):
+        return PDFURLBuilder(
+            google_drive_api,
+            secure_link_service,
+            pyramid_request.route_url,
+            NGINXSigner,
```

Comment:
```suggestion
            NGINXSigner.return_value,
```
I'm confused why this doesn't cause a problem. Shouldn't this pass an instance instead of the mocked class?